catalog/transformer/catalog-transformer-streaming-lib/src/main/java/org/codice/ddf/transformer/xml/streaming/lib/XmlInputTransformer.java

```diff
@@ -260,9 +269,37 @@ public MetacardType getMetacardType() {
             }
         }
         attributeDescriptors.addAll(BasicTypes.BASIC_METACARD.getAttributeDescriptors());
-        return new DynamicMetacardType(attributeDescriptors, id);
+
+        DynamicMetacardType dynamicMetacardType = new DynamicMetacardType(attributeDescriptors, id);
+        registerMetacardType(dynamicMetacardType);
+        return dynamicMetacardType;
+    }
+
+    private void registerMetacardType(MetacardType metacardType) {
```

Comment:
Can multiple threads be invoking this transformer at once? If so, all access to the `metacardTypeServiceRegistrations` map and the un/registration logic needs to be synchronized/guarded.

Also...you're reregistering on every call to transform in inputStream. That seems a bit expensive and problematic.